packages/flutter/lib/src/widgets/banner.dart

```diff
@@ -178,64 +178,32 @@ class BannerPainter extends CustomPainter {
   bool hitTest(Offset position) => false;
 
   double _translationX(double width) {
-    switch (layoutDirection) {
-      case TextDirection.rtl:
-        switch (location) {
-          case BannerLocation.bottomEnd:
-            return _kBottomOffset;
-          case BannerLocation.topEnd:
-            return 0.0;
-          case BannerLocation.bottomStart:
-            return width - _kBottomOffset;
-          case BannerLocation.topStart:
-            return width;
-        }
-      case TextDirection.ltr:
-        switch (location) {
-          case BannerLocation.bottomEnd:
-            return width - _kBottomOffset;
-          case BannerLocation.topEnd:
-            return width;
-          case BannerLocation.bottomStart:
-            return _kBottomOffset;
-          case BannerLocation.topStart:
-            return 0.0;
-        }
-    }
+    return switch ((layoutDirection, location)) {
+      (TextDirection.rtl, BannerLocation.topEnd)      => 0.0,
+      (TextDirection.ltr, BannerLocation.topStart)    => 0.0,
+      (TextDirection.rtl, BannerLocation.bottomEnd)   => _kBottomOffset,
+      (TextDirection.ltr, BannerLocation.bottomStart) => _kBottomOffset,
+      (TextDirection.rtl, BannerLocation.bottomStart) => width - _kBottomOffset,
+      (TextDirection.ltr, BannerLocation.bottomEnd)   => width - _kBottomOffset,
```

Comment:
I found it confusing that these are grouped by the right-hand value. I think sorting this as follow would make it much clearer that rtl and ltr are different depending on the BannerLocation value.

```suggestion
      (TextDirection.rtl, BannerLocation.bottomEnd)   => _kBottomOffset,
      (TextDirection.ltr, BannerLocation.bottomEnd)   => width - _kBottomOffset,
      (TextDirection.rtl, BannerLocation.bottomStart) => width - _kBottomOffset,
      (TextDirection.ltr, BannerLocation.bottomStart) => _kBottomOffset,
```